moped-editor/src/views/projects/projectView/NoteInput.js

```diff
@@ -0,0 +1,236 @@
+import { $generateHtmlFromNodes, $generateNodesFromDOM } from '@lexical/html';
+import { $getRoot } from 'lexical';
+import { LinkPlugin } from '@lexical/react/LexicalLinkPlugin';
+import { LinkNode } from '@lexical/link';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "100%",
+    backgroundColor: theme.palette.background.paper,
+  },
+  cancelButton: {
+    margin: theme.spacing(1),
+    position: "relative",
+    color: theme.palette.secondary.dark,
+  },
+  contentEditable: {
+    height: "300px",
+    width: "100%",
+    padding: "8px 8px",
+    border: "1px solid #fff"
+  },
+  placeholder: {
+    position: "absolute",
+    top: "24px",
+    paddingLeft: "8px"
+  },
+  editorWrapper: {
+    position: "relative"
+  },
+  showButtonItem: {
+    margin: theme.spacing(2),
+  },
+}));
+
+/**
+ * Defines the NoteTypeRadioButtons that allow a user to select the note type of a
+ * note they are creating or editing.
+ * @param {integer} defaultValue - The note type id that determines which button option
+ * should be selected/highlighted by default
+ * @param {function} onChange - Callback function that runs when the note type is changed via button click
+ * @return {JSX.Element}
+ * @constructor
+ */
+const NoteTypeRadioButtons = ({ defaultValue, onChange }) => (
+  <RadioGroup
+    row
+    value={defaultValue}
+    onChange={onChange}
+    sx={{ color: "black" }}
+  >
+    <FormControlLabel value={1} control={<Radio />} label="Internal note" />
+    <FormControlLabel value={2} control={<Radio />} label="Status update" />
+  </RadioGroup>
+);
+
+const theme = {};
+
+// Catch any errors that occur during Lexical updates and log them
+// or throw them as needed.
+const onError = (error) => {
+  console.error(error);
+}
+
+// On change, return editor content as HTML
+const OnChangePlugin = ({ onChange }) => {
+  const [editor] = useLexicalComposerContext();
+  useEffect(() => {
+    return editor.registerUpdateListener(({ editorState }) => {
+      editorState.read(() => {
+        const htmlContent = $generateHtmlFromNodes(editor, null);
+        onChange(htmlContent);
+      });
+    });
+  });
+  return null;
+}
+
+// On successful save, clear and refocus the editor
+const OnSavePlugin = ({ noteAddSuccess }) => {
+  const [editor] = useLexicalComposerContext();
+  useEffect(() => {
+    noteAddSuccess && editor.update(() => {
+      const root = $getRoot();
+      root.clear();
+    });
+    editor.focus();
+  }, [noteAddSuccess, editor]);
+  return null;
+}
+
+// On edit, load the selected comment and refocus the editor
+const OnEditPlugin = ({ htmlContent, editingNote }) => {
+  const [editor] = useLexicalComposerContext();
+  useEffect(() => {
+    const editorState = editor.getEditorState()
+    const nodeMap = editorState._nodeMap;
+    const textArray = [...nodeMap.entries()][2];
+    const isTextInEditor = !!textArray;
+    const isTextinDatabase = !!htmlContent.replace(/<[^>]*>/g, "");
```

Comment:
i'm trying to interpret what's going on here. i guess this regex is stripping out HTML characters in case someone saved a comment with hidden html?

Reply:
I think that is the case here. We use the same regex and `replace` call to disable the save button so we don't save empty HTML tags into the database. This was a Quill quirk so I wonder if we have the same issue with Lexical. Tilly, could you check if there is a graceful way to handle this in Lexical? If not, it would be awesome to pull out the two uses of this check into a reusable helper function and add a docstring to it.

Here is that comment:
`// disable save button if no text after removing html tags`

Reply:
after a bit of digging, I found a Lexical method called `$isRootTextContentEmpty` that checks whether a root node is empty, so I added that check _before_ serializing the node into HTML, so it will not be possible to save an empty tag into the db and I was able to drop the regex. thanks for pointing this out!